datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py

```diff
@@ -146,26 +147,59 @@ def __init__(
         self._runSampleEvent = Event()
         self._sampleCompleteEvent = Event()
 
-        thread = Thread(target=self._query_sample_loop, name=class_name)
-        thread.daemon = True
+    def __enter__(self):
+        """
+        Start internal thread for sampling
+        """
+        thread = Thread(target=self._query_sample_loop, name=self.class_name, daemon=True)
```

Comment:
```suggestion
        thread = Thread(target=self._query_sample_loop, name=self.class_name)
        thread.daemon = True  # Python 2 does not support daemon as Thread constructor parameter
```

Reply:
Fix for https://dev.azure.com/datadoghq/integrations-core/_build/results?buildId=9200&view=logs&j=75c19edb-262a-5378-7a8e-a2d5786f265d&t=b941f000-3f86-5ab6-68b3-41fcc47443b0&l=115